lib/syslog/transformer/syslog-decoder.js

```diff
@@ -26,7 +26,9 @@ class SyslogDecoder extends Stream.Transform {
       // Push the data when new line is sent
       if (data[i] === NEW_LINE_CODE) {
         if (this.bufferIndex !== 0) {
-          this.push(this.buffer.toString('utf8', 0, this.bufferIndex), 'utf8');
+          const stringBuffer = Buffer.allocUnsafe(this.bufferIndex);
+          this.buffer.copy(stringBuffer, 0, 0, this.bufferIndex);
+          this.push(stringBuffer.toString(), 'utf8');
```

Comment:
should we explicitly set the encoding for toString() ?

Reply:
I can. It shouldn't hurt